jl4/src/L4/Syntax.hs

```diff
@@ -115,7 +115,12 @@ data GivethSig n =
   deriving anyclass (SOP.Generic, ToExpr, NFData)
 
 data Decide n =
-  MkDecide Anno (TypeSig n) (AppForm n) (Expr n)
+  MkDecide Anno (TypeSig n) (AppFormAka n) (Expr n)
+  deriving stock (GHC.Generic, Eq, Show)
+  deriving anyclass (SOP.Generic, ToExpr, NFData)
+
+data AppFormAka n =
+  MkAppFormAka Anno (AppForm n) (Maybe (Aka n))
```

Comment:
what is the reason for not merging AppForm and AppFormAka? Are there constructs that have an AppForm but no possibility to add an aka?

Reply:
This is a good idea. I'll try to do it.